src/masks/InteractionMasks.tsx

```diff
@@ -256,7 +256,14 @@ export default function InteractionMasks<R, K extends keyof R>({
 
   function changeCellFromEvent(e: React.KeyboardEvent<HTMLDivElement>): void {
     e.preventDefault();
-    const nextPosition = getNextPosition(e.keyCode);
+    let nextPosition = getNextPosition(e.keyCode);
+    nextPosition = getNextSelectedCellPosition<R>({
```

Comment:
This is the old behavior. I missed it in  https://github.com/adazzle/react-data-grid/pull/1860/files#diff-63a962509ad28f9bf4dbc6c8a1c50569L384